utils/edit-ro-crate.py

```
@@ -0,0 +1,457 @@
+        "encodingFormat": "application/zip"
+    },
+    {
+        "@id": "5_8S.fa.gz",
+        "@type": "File",
+        "name": "Sequences mapping to ribonucleus 5_8S rRNA gene.",
+        "encodingFormat": "application/zip"
+    },
+    {
+        "@id": "alpha_tmRNA.RF01849.fasta.gz",
+        "@type": "File",
+        "name": "Sequences mapping to alpha transfer-messenger RNA.",
+        "encodingFormat": "application/zip"
+    },
+    {
+        "@id": "Bacteria_large_SRP.RF01854.fasta.gz",
+        "@type": "File",
+        "name": "Sequences mapping to bacterial large signal recognition particle RNAs.",
+        "encodingFormat": "application/zip"
+    },
+    {
+        "@id": "Bacteria_small_SRP.RF00169.fasta.gz",
+        "@type": "File",
+        "name": "Sequences mapping to bacterial small signal recognition particle RNAs.",
+        "encodingFormat": "application/zip"
+    },
+    {
+        "@id": ".merged_CDS.faa",
+        "@type": "File",
+        "name": "Coding sequences with amino acids.",
+        "encodingFormat": "text/plain"
+    },
+    {
+        "@id": ".merged_CDS.ffn",
+        "@type": "File",
+        "name": "Coding sequences with nucleotides.",
+        "encodingFormat": "text/plain"
+    }
+]
+
+def main(target_directory, extended_config_yaml, ena_run_accession_id, metagoflow_version ):
+    """
+    Edit the output of the `rocrate init` tool. 
+    Map files descriptions ; should be done through the cwl descriptions but we go that way at a later point.
+    Add extra ids on the ro-crate describing the metaGOflow workflow and the ENA data (if used). 
+    """
+
+    crate = ROCrate(target_directory) # here we use a complete directory, could we use just the json..?
+
+    for entry in crate.get_entities():
+        for description in descriptions:
+            try:
+                pk = description["@id"] in entry.id
+            except:
+                pass
```

we could throw the exception here and print the stack trace

suggestion:
import traceback
e.g traceback.print_exc()